plotly/plotly/plotly.py

```diff
@@ -1520,6 +1520,77 @@ def get_dashboard_names(cls):
         return [str(dboard['filename']) for dboard in dashboards]
 
 
+class presentation_ops:
+    """
+    Interface to Plotly's Spectacle-Presentations API.
+    """
+    @classmethod
+    def upload(cls, presentation, filename, sharing='public', auto_open=True):
+        """
+        Function for uploading presentations to Plotly.
+
+        :param (dict) presentation: the JSON presentation to be uploaded. Use
+            plotly.presentation_objs.Presentation to create the presentation
+            from a string.
+        :param (str) filename: the name of the presentation to be saved in
+            your Plotly account. Will overwrite a presentation of the same
+            name if it already exists in your files.
+        :param (str) sharing: can be set to either 'public', 'private'
+            or 'secret'. If 'public', your presentation will be viewable by
+            all other users. If 'private' only you can see your presentation.
+            If 'secret', the url will be returned with a sharekey appended
+            to the url. Anyone with the url may view the presentation.
+        :param (bool) auto_open: automatically opens the presentation in the
+            browser.
+
+        See the documentation online for examples.
+        """
+        if sharing == 'public':
+            world_readable = True
+        elif sharing == 'private':
+            world_readable = False
+        elif sharing == 'secret':
+            world_readable = False
+
```

Comment:
like in other wrappers, we should raise an exception if `sharing` wasn't one of those strings